### PR TITLE
feat: cypress tests for list v2

### LINKED ISSUE #17901: Task: List v2 - Cypress tests
Add new tests and migrate List V1 tests to V2

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description

Write Cypress Tests for List V2
[Test Plan](https://www.notion.so/appsmith/Test-plan-async-406c12f0aeb54230b00e19d10cae6f54)

- Property Pane Tests
- Child Widgets Test
      - Select Widgets
      - Input Widgets
      - Radio Widget
      - Modal Widget
      - Checkbox group
      - Text Widget
      - File Picker

Fixes #17901

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/cypress/locators/commonlocators.json [hunk 1]
```diff
@@ -206,7 +208,9 @@
...
   "generalSectionHeight": ".t--property-pane-section-general .t--property-control-label:contains('Height')",
-  "listPaginateNextButton":".t--widget-listwidgetv2 .rc-pagination-next .rc-pagination-item-link",
+  "listPaginateNextButton": ".t--widget-listwidgetv2 .rc-pagination-next .rc-pagination-item-link",
+  "listPaginatePrevButton": ".t--widget-listwidgetv2 .rc-pagination-prev .rc-pagination-item-link",
   "PropertyPaneSearchInput": ".t--property-pane-search-input-wrapper input",
-  "BorderRadius0px": ".t--property-control-borderradius .t--button-tab-0px"
-}
+  "BorderRadius0px": ".t--property-control-borderradius .t--button-tab-0px",
+  "EnableFormatting": ".t--property-control-enableformatting input[type='checkbox']"
+}
```
Comment (ashit-rath) [inline app/client/cypress/locators/commonlocators.json:216]: Missing new line

File: app/client/src/widgets/ListWidgetV2/index.ts [hunk 2]
```diff
@@ -105,7 +105,7 @@ export const CONFIG = {
                       RegisteredWidgetFeatures.DYNAMIC_HEIGHT,
                     ],
                     shouldScrollContents: false,
-                    dynamicHeight: "FIXED",
+                    // dynamicHeight: "FIXED",
```
Comment (ashit-rath) [inline app/client/src/widgets/ListWidgetV2/index.ts:108]: Are we suppose to remove it?

File: app/client/src/utils/autocomplete/EntityDefinitions.ts [hunk 3]
```diff
@@ -208,6 +208,8 @@ export const entityDefinitions = {
       "!url": "https://docs.appsmith.com/widget-reference/dropdown",
     },
     isDisabled: "bool",
+    isValid: "bool",
```
Comment (ashit-rath) [inline app/client/src/utils/autocomplete/EntityDefinitions.ts:211]: Why are we adding these properties in this PR?

Comment (Tooluloope) [reply app/client/src/utils/autocomplete/EntityDefinitions.ts:211]: Users don't have access to them in the List widget but can access them normally when dropped on the canvas.
`currentItemsView` only has properties of the widget defined in the `entityDefinition`.

File: app/client/src/widgets/ListWidgetV2/index.ts (hunk 2 above)
Comment (Tooluloope) [reply app/client/src/widgets/ListWidgetV2/index.ts:108]: Yeah, that's the only way I could drop widgets on the container

**Review by ashit-rath — APPROVED** [review]

**Pull request merged**